rust/connlib/tunnel/src/tests.rs

```diff
@@ -2367,24 +2360,6 @@ fn assert_icmp_packets_properties(state: &mut TunnelTest, ref_state: &ReferenceS
                     possible_resource_ips.contains(&actual_destination),
                     "ICMP request on gateway to target a known resource IP"
                 );
-
-                match state
-                    .client_proxy_ip_mapping
-                    .entry(client_sent_request.destination())
-                {
-                    Entry::Vacant(v) => {
-                        // We have to gradually discover connlib's mapping ...
-                        // For the first packet, we just save the IP that we ended up talking to.
-                        v.insert(gateway_received_request.destination());
-                    }
-                    Entry::Occupied(o) => {
-                        assert_eq!(
-                            gateway_received_request.destination(),
-                            *o.get(),
-                            "ICMP request on client to target correct same IP of DNS resource"
-                        );
-                    }
-                }
```

Comment:
we no longer test the stability of the mapping?

Reply:
Hmm, let me think why I removed that.

Reply:
I guess I can still rely on the order to find the originally sent request and ensure the mapping is consistent?

Reply:
yeah, I think that would be good to keep